tests/socket/socket_tests.cpp

```diff
@@ -28,6 +28,12 @@ using namespace std::chrono_literals;
 #include <mstcpip.h>
 
 CATCH_REGISTER_LISTENER(_watchdog)
+typedef struct _test_globals
+{
+    bpf_object_ptr bpf_object;
+} test_globals_t;
+
+static test_globals_t _globals;
```

Comment:
why a global instead of a bpf_object_ptr that is local for each test case?
I suspect a global would show up as a leak if the usersim library were initialized/terminated for each test case to do leak detection.

Reply:
Moved `test_globals_t` to the `common_tests.h` header file to be used for multiple tests.

Reply:
As a reminder since it's been several days, I am still waiting for my comment to be addressed.   Moving a global to another file (which I don't see in this PR either) does not address my comment about the fact that I don't think it should be a global.